strutil/pathiter.go

```diff
@@ -66,60 +66,73 @@ func (iter *PathIterator) Path() string {
 	return iter.path
 }
 
-// CurrentBase returns the name of the current path element.
-// It will only end with a slash if the iterator is currently
-// placed at the base of the filesystem.
-// Ex:
+// CurrentBase returns the last element of the current path.
+// It removes any trailing slash unless the last element of the
+// current path is the filesystem root (and so CurrentBase() == "/").
+// If the current path is empty (i.e. Next() has not yet been called),
+// so is CurrentBase.
+// Example:
 //
-//	iter:= NewPathIterator("/foo/bar")
+//	iter:= NewPathIterator("/foo/bar")  // iter.CurrentBase() == ""
 //	iter.Next() // iter.CurrentBase() == "/" && iter.CurrentPath() == "/"
 //	iter.Next() // iter.CurrentBase() == "foo" && iter.CurrentPath() == "/foo"
 //	iter.Next() // iter.CurrentBase() == "bar" && iter.CurrentPath() == "/foo/bar"
 func (iter *PathIterator) CurrentBase() string {
-	// Only remove ending slash if the iterator is not pointing to the base of the filesystem
+	// Only remove trailing slash if the iterator is not pointing to the filesystem root
 	if iter.right > 1 && iter.path[iter.right-1:iter.right] == "/" {
 		return iter.path[iter.left : iter.right-1]
 	}
 	return iter.path[iter.left:iter.right]
 }
 
-// CurrentPath returns the prefix of path that was traversed, including the current base.
+// CurrentPath returns the path up to the current depth.
+// If the depth is 0 (i.e. Next has not yet been called),
```

Comment:
Next vs Next() -> not consistent with how we reference methods, we typically do not include the `()`.